mobile/src/home-view.js

```diff
@@ -66,40 +66,43 @@ class HomeView extends PureComponent {
         this.loadLocalSurveys()
         const survRef = fbc.database.public.adminRef('surveys')
         const resultsRef = fbc.database.private.adminableUserRef('results')
-        resultsRef.on('child_added', data => {
-          this.setState(({ results }) => ({ results: [...results, data.key] }))
-        })
-        survRef.on('value', data => {
-          const today = new Date().getTime()
-          const surveys = Object.entries(data.val() || {})
-            .map(([key, val]) => ({ ...val, key }))
-            .filter(survey => {
-              if (survey.publishDate) return survey.publishDate < today
-              return true
-            })
-          if (surveyId) {
-            const directSurvey = surveys.find(survey => survey.key === surveyId)
-            if (directSurvey) {
-              this.selectSurvey(directSurvey)
-              this.setState({ showTable: false })
+        resultsRef.on('value', data => {
+          const results = Object.entries(data.val() || {}).map(([key, val]) => key)
+          this.setState({ results })
+          survRef.on('value', data => {
```

Comment:
This will wire a new handler every time resultsRef's value changes

Reply:
Would be doing an ".off" before the new handler be the best solution?

Reply:
maybe, but in this case, it looks like we are not actually looking at a different `survRef`key if resultsRef changes.  It looks like there is some derived state that depends on the combination of both results and surveys in firebase.  But there is no need to re-fetch the surveys from firebase if results change.  Trying to re-fetch and unwire/wire handlers will also be tricky and error prone.  The right thing will be to set state directly based on firebase values in `resultsRef` and `survRef` (which will trigger re-renders when either of these changes) and compute all the derived data from these 2 bits of state during the render.  If this is expensive, it can be memoized.  `memoize-one` is perfect for this. https://www.npmjs.com/package/memoize-one